src/chunk.c

```diff
@@ -880,6 +993,193 @@ chunk_find_all_oids(Hyperspace *hs, List *dimension_vecs, LOCKMODE lockmode)
 	return oid_list;
 }
 
+Datum
+chunk_show(PG_FUNCTION_ARGS)
+{
+	FuncCallContext *funcctx;
+	int			call_cntr;
+	int			max_calls;
+	TupleDesc	tupdesc;
+	AttInMetadata *attinmeta;
+
+	List	   *result_set = NIL;
+
+	/* stuff done only on the first call of the function */
+	if (SRF_IS_FIRSTCALL())
+	{
+		Cache	   *hypertable_cache;
```

Comment:
I would break out the Logic that returns the List of chunks into a separate function so it's not interspersed with the SRF bullshit. This function can then just return a List that would populate funcctx->user_fctx. It would take table_relid, older_than, newer_than (in the internal time representation). Then the remaining shell could deal with the SRF stuff + type checking.